src/rabbit_amqqueue.erl

```diff
@@ -1245,9 +1245,17 @@ delete_crashed_internal(Q, ActingUser) ->
 
 purge(Q) when ?amqqueue_is_classic(Q) ->
     QPid = amqqueue:get_pid(Q),
+    ok = rabbit_event:notify(queue_purged,
+                             [{name, amqqueue:get_name(Q)},
+                              {user_who_performed_action, amqqueue:get_user(Q)},
```

Comment:
Are you sure this function exists? What would a user associated with a queue mean?

If you take a look at how the `queue.delete` handler obtains the acting user, it would be quite different: the value is passed on from the connection/channel. `rabbit_amqueue:purge/1` has to become `rabbit_amqquue:purge/2` and accept an acting user as a parameter. Which will require updating all call sites, including those in the management plugin and CLI tools.